modules/Oculus_module/src/OculusModule.cpp

```diff
@@ -518,7 +531,10 @@ bool OculusModule::configure(yarp::os::ResourceFinder& rf)
         yInfo() << "[OculusModule::configure] Cameras have been reset.";
     }
 
-    m_state = OculusFSM::Configured;
+    {
+        std::lock_guard<std::mutex> guard(m_mutex);
```

Comment:
I would lock the entire method. ``m_state`` is not the only variable that could be affected by concurrent read/write